openstack/baremetal/v1/nodes/results.go

```diff
@@ -48,6 +48,20 @@ func ExtractNodesInto(r pagination.Page, v interface{}) error {
 	return r.(NodePage).Result.ExtractIntoSlicePtr(v, "nodes")
 }
 
+// Extract interprets a BIOSSettingsResult as an array of BIOSSetting structs, if possible.
+func (r BIOSSettingsResult) Extract() (*[]BIOSSetting, error) {
+	var s BIOSSettings
+	err := r.ExtractInto(&s)
+	return &s.Settings, err
+}
```

Comment:
What you have works, but I think the following version more closely matches other functions within Gophercloud:

```go
// Extract interprets a BIOSSettingsResult as an array of BIOSSetting structs, if possible.
func (r BIOSSettingsResult) Extract() ([]BIOSSetting, error) {
  var s struct {
    Settings []BIOSSetting `json:"bios"`
  }
  err := r.ExtractInto(&s)
  return s.Settings, err
}
```

You can then remove the `BIOSSettings` type as it's no longer used.

This change will require an update to the unit tests, but it's a small update.

Reply:
Thanks, that's a good change.  Updated.